### PR TITLE
acrn-tools: adjust life_mngr patch to reflect latest changes upstream

### PR DESCRIPTION
**Pull request by gvancuts**
Update patch used to not build 'life_mngr' by default to reflect the
latest Makefile changes upstream.

Signed-off-by: Geoffroy Van Cutsem <geoffroy.vancutsem@intel.com>

### CODE REVIEW

File: recipes-core/acrn/acrn-tools/0001-meta-acrn-do-not-build-life_mngr-by-default.patch [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Date: Wed, 3 Mar 2021 10:24:38 +0100
+Subject: [PATCH] meta-acrn: do not build life_mngr by default
+
+Do not build (or install) the ACRN life_mngr by default as this is
+a User VM tool, not one to be used and run in the Service VM.
+
+The component can still be built independantly by invoking
+'make -C misc life_mngr' (components will be built and placed in
+'misc/build/services/' by default).
+
```
Comment (saininav) [inline recipes-core/acrn/acrn-tools/0001-meta-acrn-do-not-build-life_mngr-by-default.patch:12]: Could you please add Upstream-status to this patch  ? 

Upstream-Status: Inappropriate [Build Specific or other reason]

Ref: https://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines

**Comment by wenlingz** [comment]
@saininav  geoffroy  need create this PR on master-next ?

File: recipes-core/acrn/acrn-tools/0001-meta-acrn-do-not-build-life_mngr-by-default.patch (hunk 1 above)
Comment (gvancuts) [reply recipes-core/acrn/acrn-tools/0001-meta-acrn-do-not-build-life_mngr-by-default.patch]: > Could you please add Upstream-status to this patch ?

Done. I used `Pending` though as I believe this should be submitted upstream. And it is my intention to do it ;-)

Comment (gvancuts) [reply recipes-core/acrn/acrn-tools/0001-meta-acrn-do-not-build-life_mngr-by-default.patch]: > Done. I used `Pending` though as I believe this should be submitted upstream. And it is my intention to do it ;-)

FYI - https://github.com/projectacrn/acrn-hypervisor/pull/5814

**Comment by wenlingz** [comment]
@gvancuts your PR still has build issue(DM), please help modify it, thanks.please refer the log
64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/include -I/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/recipe-sysroot/usr/include/acrn/services -DASL_COMPILER=\"/usr/bin/iasl\" -fstack-protector-strong -DDM_DEBUG -c log/disk_logger.c -o /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/log/disk_logger.o -MMD -MT /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/log/disk_logger.o
| core/monitor.c:23:10: fatal error: acrn_mngr.h: No such file or directory
|    23 | #include "acrn_mngr.h"
|       |          ^~~~~~~~~~~~~
| compilation terminated.
| Makefile:243: recipe for target '/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/core/monitor.o' failed
| make[1]: *** [/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/core/monitor.o] Error 1
| make[1]: *** Waiting for unfinished jobs....
| make[1]: Leaving directory '/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/git/devicemodel'
| Makefile:117: recipe for target 'devicemodel' failed
| make: *** [devicemodel] Error 2
| WARNING: /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/temp/run.do_compile.1489:1 exit 1 from 'exit 1'
| 
NOTE: recipe acrn-devicemodel-5810-1614916878: task do_compile: Failed
NOTE: recipe acrn-hypervisor-5810-1614916878: task do_configure: Succeeded
ERROR: Task (/home/meta-acrn/workspace/meta-acrn/recipes-core/acrn/acrn-devicemodel.bb:do_compile) failed with exit code '1'
NOTE: recipe acrn-tools-5810-1614916878: task do_package: Succeeded
NOTE: Tasks Summary: Attempted 2025 tasks of which 1969 didn't need to be rerun and 1 failed.

**Comment by gvancuts** [comment]
> 
> 
> @gvancuts your PR still has build issue(DM), please help modify it, thanks.please refer the log
> 64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/include -I/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/recipe-sysroot/usr/include/acrn/services -DASL_COMPILER="/usr/bin/iasl" -fstack-protector-strong -DDM_DEBUG -c log/disk_logger.c -o /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/log/disk_logger.o -MMD -MT /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/log/disk_logger.o
> | core/monitor.c:23:10: fatal error: acrn_mngr.h: No such file or directory
> | 23 | #include "acrn_mngr.h"
> | | ^~~~~~~~~~~~~
> | compilation terminated.
> | Makefile:243: recipe for target '/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/core/monitor.o' failed
> | make[1]: *** [/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/core/monitor.o] Error 1
> | make[1]: *** Waiting for unfinished jobs....
> | make[1]: Leaving directory '/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/git/devicemodel'
> | Makefile:117: recipe for target 'devicemodel' failed
> | make: *** [devicemodel] Error 2
> | WARNING: /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/temp/run.do_compile.1489:1 exit 1 from 'exit 1'
> |
> NOTE: recipe acrn-devicemodel-5810-1614916878: task do_compile: Failed
> NOTE: recipe acrn-hypervisor-5810-1614916878: task do_configure: Succeeded
> ERROR: Task (/home/meta-acrn/workspace/meta-acrn/recipes-core/acrn/acrn-devicemodel.bb:do_compile) failed with exit code '1'
> NOTE: recipe acrn-tools-5810-1614916878: task do_package: Succeeded
> NOTE: Tasks Summary: Attempted 2025 tasks of which 1969 didn't need to be rerun and 1 failed.

I have seen this error before, and it's solved by this part of the patch upstream: https://github.com/projectacrn/acrn-hypervisor/pull/5810/files#diff-9a873d0654d993493b5abfa0b5c53f1378404b48da44c2e81e184c46cfe0229c

Now, I noticed that you carry a patch in Yocto that removes the `tools` dependency from the `devicemodel` target ([here](https://github.com/intel/meta-acrn/blob/master/recipes-core/acrn/acrn-devicemodel/dont-build-tools.patch)). This is not correct, and I'm actually surprised that you've never seen this error before. If I apply this patch (i.e. remove the `devicemodel` dependency on `tools`) to my `acrn-hypervisor` tree (even without the patches from this PR), I get the same failure that you report above. Why have you not seen this before (and outside of this patch)?

**Comment by saininav** [comment]
> > @gvancuts your PR still has build issue(DM), please help modify it, thanks.please refer the log
> > 64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/include -I/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/recipe-sysroot/usr/include/acrn/services -DASL_COMPILER="/usr/bin/iasl" -fstack-protector-strong -DDM_DEBUG -c log/disk_logger.c -o /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/log/disk_logger.o -MMD -MT /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/log/disk_logger.o
> > | core/monitor.c:23:10: fatal error: acrn_mngr.h: No such file or directory
> > | 23 | #include "acrn_mngr.h"
> > | | ^~~~~~~~~~~~~
> > | compilation terminated.
> > | Makefile:243: recipe for target '/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/core/monitor.o' failed
> > | make[1]: *** [/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/build/devicemodel/core/monitor.o] Error 1
> > | make[1]: *** Waiting for unfinished jobs....
> > | make[1]: Leaving directory '/home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/git/devicemodel'
> > | Makefile:117: recipe for target 'devicemodel' failed
> > | make: *** [devicemodel] Error 2
> > | WARNING: /home/meta-acrn/workspace/build/master-acrn-sos/work/corei7-64-oe-linux/acrn-devicemodel/5810-1614916878/temp/run.do_compile.1489:1 exit 1 from 'exit 1'
> > |
> > NOTE: recipe acrn-devicemodel-5810-1614916878: task do_compile: Failed
> > NOTE: recipe acrn-hypervisor-5810-1614916878: task do_configure: Succeeded
> > ERROR: Task (/home/meta-acrn/workspace/meta-acrn/recipes-core/acrn/acrn-devicemodel.bb:do_compile) failed with exit code '1'
> > NOTE: recipe acrn-tools-5810-1614916878: task do_package: Succeeded
> > NOTE: Tasks Summary: Attempted 2025 tasks of which 1969 didn't need to be rerun and 1 failed.
> 
> I have seen this error before, and it's solved by this part of the patch upstream: https://github.com/projectacrn/acrn-hypervisor/pull/5810/files#diff-9a873d0654d993493b5abfa0b5c53f1378404b48da44c2e81e184c46cfe0229c
> 
> Now, I noticed that you carry a patch in Yocto that removes the `tools` dependency from the `devicemodel` target ([here](https://github.com/intel/meta-acrn/blob/master/recipes-core/acrn/acrn-devicemodel/dont-build-tools.patch)). This is not correct, and I'm actually surprised that you've never seen this error before. If I apply this patch (i.e. remove the `devicemodel` dependency on `tools`) to my `acrn-hypervisor` tree (even without the patches from this PR), I get the same failure that you report above. Why have you not seen this before (and outside of this patch)?

In current meta-acrn implementation, tools are built with separate acrn-tools recipe. And to build devicemodel, acrn-tools are added to DEPENDS, so no need to build tools again at the time of devicemodel build. 
https://github.com/intel/meta-acrn/blob/master-next/recipes-core/acrn/acrn-devicemodel.bb#L7

Seem acrn-tools not providing required dependencies (headers) ?

**Comment by saininav** [comment]
Previously CFLAGS += -I$(TOOLS_OUT) allowed to add path of acrn-tools in acrn-devicemodel recipe. But with this PR it can't.

https://github.com/intel/meta-acrn/blob/master-next/recipes-core/acrn/acrn-devicemodel.bb#L10

**Comment by gvancuts** [comment]
> In current meta-acrn implementation, tools are built with separate acrn-tools recipe. And to build devicemodel, acrn-tools are added to DEPENDS, so no need to build tools again at the time of devicemodel build.
> https://github.com/intel/meta-acrn/blob/master-next/recipes-core/acrn/acrn-devicemodel.bb#L7

I see - thanks for the explanation!

> 
> Seem acrn-tools not providing required dependencies (headers) ?

It is, but instead of putting them under `$(TOOLS_OUT)` like it did before, it's putting them under `$(TOOLS_OUT)/services`.

This is what the upstream patch does (amongst other things)... but I'm not too sure how to handle this in this `meta-acrn` layer, can you guide me?

**Comment by gvancuts** [comment]
> Previously CFLAGS += -I$(TOOLS_OUT) allowed to add path of acrn-tools in acrn-devicemodel recipe. But with this PR it can't.

This is why this PR modify both the `CFLAGS` and LDFLAGS` to use ~$(TOOLS_OUT)/services` instead. But it seems this is not picked up when building with Yocto, why?

**Comment by saininav** [comment]
> > In current meta-acrn implementation, tools are built with separate acrn-tools recipe. And to build devicemodel, acrn-tools are added to DEPENDS, so no need to build tools again at the time of devicemodel build.
> > https://github.com/intel/meta-acrn/blob/master-next/recipes-core/acrn/acrn-devicemodel.bb#L7
> 
> I see - thanks for the explanation!
> 
> > Seem acrn-tools not providing required dependencies (headers) ?
> 
> It is, but instead of putting them under `$(TOOLS_OUT)` like it did before, it's putting them under `$(TOOLS_OUT)/services`.
> 
> This is what the upstream patch does (amongst other things)... but I'm not too sure how to handle this in this `meta-acrn` layer, can you guide me?

Can you introduce some variable to CFLAGS AND LDFLAGS which will allow Yocto to pass acrn-tools artifacts path !

Otherwise it is difficult to pass without overwriting Makefile's CFLAGS and LDFLAGS, which we do not want !

**Comment by saininav** [comment]
@wenlingz Could you test  PR#5810  with https://github.com/saininav/meta-acrn/commit/8f1649ed364bddc07ffaf61def2346249e6dc53d meta-acrn patch.

**Comment by wenlingz** [comment]
> @wenlingz Could you test PR#5810 with [saininav@8f1649e](https://github.com/saininav/meta-acrn/commit/8f1649ed364bddc07ffaf61def2346249e6dc53d) meta-acrn patch.

It's PASS now

**Comment by saininav** [comment]
This patch should be for master-next, as master-next always points to latest and greatest.  I will raise new PR, whenever needed. Closing this now.

**Pull request closed without merging**